src/main/java/com/google/api/codegen/gapic/MainGapicProviderFactory.java

```diff
@@ -139,6 +144,22 @@
 
       return Arrays.<GapicProvider<? extends Object>>asList(mainProvider, packageInfoProvider);
 
+    } else if (id.equals(JAVA_TEST)) {
+      GapicCodePathMapper javaPathMapper =
+          CommonGapicCodePathMapper.newBuilder()
+              .setPrefix("src/test/java")
```

Comment:
Nit: Maybe use a path builder of some sort, rather than hardcoding path separators?